### PR TITLE
expand the contributor guide

### LINKED ISSUE #1694: Expand Contributing doc
We should have a dedicated Contributing doc that is much more inviting for potential contribs.  Currently, https://v2--brigade-docs.netlify.app/topics/contributing/ just links back to the community Contributing.md which is more around signing commits and formalities.

A few thoughts:

 - Add a link to issues in the main brigade repo tagged 'good-first-issue' (e.g. https://github.com/brigadecore/brigade/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22)
 - Mention/link to auxiliary repos that also are ripe for outside contributors (gateways? sdks?)
 - Link to the [QuickStart](https://v2--brigade-docs.netlify.app/intro/quickstart/) and/or [Developers](https://v2--brigade-docs.netlify.app/topics/developers/) docs
 - Other/more?

### PR DESCRIPTION
**Pull request by krancour**
Fixes #1694

It is also my intention that new and very short contributor guides for all peripheral components will include only component-specific instruction and will link back to here to discuss our general development approach.

### CODE REVIEW

File: docs/content/topics/contributor-guide/workflow.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+## Tips for Opening a Successful PR
+
+* Make sure your PR's description clearly and succinctly summarizes the changes
+  proposed by your PR.
+
+* Resist the temptation to bundle _unrelated_ changes into a single PR.
+  Discrete PRs are easier for maintainers to review and sign off on.
+
+* A larger PRs is more likely to be merged if it references and address
```
Comment (vdice) [inline docs/content/topics/contributor-guide/workflow.md:215]:
```suggestion
* A larger PR is more likely to be merged if it references and addresses
```

File: docs/content/topics/contributor-guide/hacking.md [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+
+To destroy your KinD cluster, use `make hack-kind-down`.
+
+> ⚠️&nbsp;&nbsp;`make hack-kind-down` deliberately leaves your local registry
+> running so that if you resume work later, you are doing so with a local
+> registry that's already primed with most layers of each of Brigade's images.
+> If you wish to destroy the registry, use:
+>
+> ```shell
+> $ docker rm -f brigade-github-gateway-dev-cluster-control-plane
```
Comment (vdice) [inline docs/content/topics/contributor-guide/hacking.md:280]: Should this be `brigade-dev-registry`?

**Review by vdice — APPROVED** [review]
Noticed a few minor items, otherwise LGTM!  Excellent doc additions.

File: docs/content/topics/contributor-guide/hacking.md (hunk 2 above)
Comment (krancour) [reply docs/content/topics/contributor-guide/hacking.md:280]: It should be, but I also copied/pasted that directly from my terminal, so I may need to see if something isn't working as I had intended.

I thought I had this and all the peripherals set to use the _same_ local registry.

Comment (krancour) [reply docs/content/topics/contributor-guide/hacking.md:280]: Oh... I just literally copied the wrong container name after a `docker ps`. Will fix.

Comment (krancour) [reply docs/content/topics/contributor-guide/hacking.md]: Fixed

**Pull request merged**